### PR TITLE
fix: reobtain in disclosure

### PR DESCRIPTION
**Pull request by sjoerdlmkns**

### CODE REVIEW

**Review by bobhageman — APPROVED** [review]

File: integration_test/disclosure_session/special_scenarios/revocation.dart [hunk 1]
```diff
@@ -38,39 +53,69 @@ Future<void> revocationTest(WidgetTester tester, IntegrationTestIrmaBinding irma
...
 
   // The disclosure permission overview screen should be visible.
+  expect(find.byType(DisclosurePermissionChoicesScreen), findsOneWidget);
   expect(find.text('Share my data'), findsOneWidget);
   expect(find.text('Share my data with demo.privacybydesign.foundation'), findsOneWidget);
-  expect(find.text('Demo MijnOverheid.nl'), findsOneWidget);
-  expect(find.text('12345'), findsOneWidget);
-  expect(find.text('Revoked'), findsOneWidget);
 
+// Find all credential cards
```
Comment (sanderhollaar) [inline integration_test/disclosure_session/special_scenarios/revocation.dart:60]: leading whitespace

**Review by ivard — APPROVED** [review]

File: integration_test/disclosure_session/special_scenarios/revocation.dart [hunk 2]
```diff
@@ -38,39 +53,69 @@ Future<void> revocationTest(WidgetTester tester, IntegrationTestIrmaBinding irma
...
-        )
-        .onPressed,
+    tester.widget<YiviThemedButton>(shareDataButtonFinder).onPressed,
     isNull,
   );
 
+  // Press change choice
+  await tester.tapAndSettle(find.text('Change choice'));
+
+  // Revoked (but obtainable) card should be visible here too.
```
Comment (sanderhollaar) [inline integration_test/disclosure_session/special_scenarios/revocation.dart:85]: trailing dot

File: integration_test/disclosure_session/special_scenarios/revocation.dart [hunk 3]
```diff
@@ -38,39 +53,69 @@ Future<void> revocationTest(WidgetTester tester, IntegrationTestIrmaBinding irma
...
+  await _evaluateDemoCredentialCard(
+    tester,
+    demoCredentialCardFinder,
+    isRevoked: true,
+  );
+
+  // Navigate to back to choices overview
+  await tester.tapAndSettle(find.byKey(const Key('irma_app_bar_leading')));
+
+  // Now reobtain the card.
```
Comment (sanderhollaar) [inline integration_test/disclosure_session/special_scenarios/revocation.dart:95]: ~Now~
trailing dot

File: integration_test/disclosure_session/special_scenarios/revocation.dart [hunk 4]
```diff
@@ -38,39 +53,69 @@ Future<void> revocationTest(WidgetTester tester, IntegrationTestIrmaBinding irma
...
 
-  expect(find.text('Revoked'), findsNothing);
+  // Revoked card should be visible here too.
+  await _evaluateDemoCredentialCard(
+    tester,
+    demoCredentialCardFinder,
+    isRevoked: false,
+  );
 
-  // Finish session.
```
Comment (sanderhollaar) [inline integration_test/disclosure_session/special_scenarios/revocation.dart:70]: trailing dot

File: integration_test/helpers/helpers.dart [hunk 5]
```diff
@@ -175,17 +182,52 @@ Future<void> evaluateCredentialCard(
...
+        );
+      }
     }
 
-    // Compare the issuer credential name
-    if (issuerName != null) {
-      expect(cardHeaderText.elementAt(1), 'Issued by:');
-      expect(cardHeaderText.elementAt(2), issuerName);
+    if (shouldCheckHeaderInfo) {
+      // Filter the status texts from the list, so we can test the rest.
```
Comment (sanderhollaar) [inline integration_test/helpers/helpers.dart:218]: trailing dot

**Review by sanderhollaar — APPROVED** [review]

**Pull request merged**